test/test_itertools.py

```diff
@@ -277,3 +279,22 @@ def test_join_items():
 
     with pytest.raises(Exception):
         oin_items(left, right, "strict")
+
+
+@pytest.mark.parametrize(
+    "iterables, probabilities, sample_size, samples",
+    [
+        ([[1, 2, 3], [4, 5, 6]], [1.0, 0.0], 5, [1, 2, 3]),
+        ([[1, 2, 3], [4, 5, 6]], [0.0, 1.0], 5, [4, 5, 6]),
```

Comment:
I don't like the test where we set the probability for an input to zero. Couldn't we initialise the random state with some seed?

Reply:
I think even if we set the seed, newer python or library versions may break the sequence generated by the same seed.

Reply:
No? The rng comes from numpy which should be fairly stable.

Reply:
No? The rng comes from numpy which should be fairly stable.